### PR TITLE
[Android] Expose onScrollChanged/onFocusChanged to XWalkView

### PR DESCRIPTION
**Pull request by XiaosongWei**
Use XWalkAPI.delegate annotation for reflection generation script
to generate delegate methods in XWalkView within which View's protected
methods is invoked.

BUG=XWALK-4899, XWALK-4895

### CODE REVIEW

**Comment by XiaosongWei** [comment]
@lincsoon @halton Please take a look, thanks!

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentView.java [hunk 1]
```diff
@@ -91,4 +92,16 @@ public boolean onTouchEvent(MotionEvent event) {
...
+    @Override
+    public void onScrollChanged(int l, int t, int oldl, int oldt) {
+        mXWalkView.onScrollChangedDelegate(l, t, oldl, oldt);
+    }
+
+    @Override
+    protected void onFocusChanged(boolean gainFocus, int direction, Rect previouslyFocusedRect) {
+        mXWalkView.onFocusChangedDelegate(gainFocus, direction, previouslyFocusedRect);
+        super.onFocusChanged(gainFocus, direction, previouslyFocusedRect);
+
```
Comment (halton) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentView.java]: nit, remove this empty line

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 2]
```diff
@@ -1287,4 +1287,17 @@ public boolean onTouchEvent(MotionEvent event) {
         return mContent.onTouchEvent(event);
     }
 
+    // Use delegate to access XWalkView's protected method
+    @XWalkAPI(delegate = true,
+              preWrapperLines = {"onScrollChanged(l, t, oldl, oldt);"})
+    public void onScrollChangedDelegate(int l, int t, int oldl, int oldt) {
+
```
Comment (halton) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: ditto

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 3]
```diff
@@ -1287,4 +1287,17 @@ public boolean onTouchEvent(MotionEvent event) {
...
+    }
+
+    @XWalkAPI(delegate = true,
+              preWrapperLines = {"onFocusChanged(gainFocus, direction, previouslyFocusedRect);"})
+    public void onFocusChangedDelegate(boolean gainFocus, int direction, Rect previouslyFocusedRect) {
+
```
Comment (halton) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: ditto

**Comment by halton** [comment]
lgtm with above nit style comments

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentView.java (hunk 1 above)
Comment (XiaosongWei) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentView.java]: Thanks, will update soon.

**Pull request merged**